src/main/java/apoc/export/json/JsonFormatSerializer.java

```diff
@@ -0,0 +1,74 @@
+package apoc.export.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.PropertyContainer;
+import org.neo4j.graphdb.Relationship;
+
+import java.io.IOException;
+
+public enum JsonFormatSerializer {
+
+    DEFAULT() {
+        @Override
+        public void writeNode(JsonGenerator jsonGenerator, Node node) throws IOException {
+
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeStringField("object_type", "node");
+            jsonGenerator.writeNumberField("id", node.getId());
+            jsonGenerator.writeArrayFieldStart("labels");
+            for (Label label : node.getLabels()) { jsonGenerator.writeString(label.toString()); }
+            jsonGenerator.writeEndArray();
+            serializerProperty(jsonGenerator, node);
+            jsonGenerator.writeEndObject();
+        }
+
+        @Override
+        public void writeRelationship(JsonGenerator jsonGenerator, Relationship relationship) throws IOException {
+
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeStringField("object_type", "relationship");
+            jsonGenerator.writeNumberField("id", relationship.getId());
+            jsonGenerator.writeStringField("type", relationship.getType().name());
+            serializerProperty(jsonGenerator, relationship);
+            jsonGenerator.writeNumberField("start_node_id", relationship.getStartNodeId());
+            jsonGenerator.writeNumberField("end_node_id", relationship.getEndNodeId());
+            jsonGenerator.writeEndObject();
+        }
+
+        @Override
+        public void writeText(JsonGenerator jsonGenerator, String name, Object value) throws IOException {
+            if(value != null) {
+                jsonGenerator.writeObjectField(name, value);
+            }
+        }
+
+        @Override
+        public void serializerProperty(JsonGenerator jsonGenerator, PropertyContainer propertyContainer) throws IOException {
+            jsonGenerator.writeObjectFieldStart("properties");
+            propertyContainer.getAllProperties().forEach((key, value)-> {
+                try {
+                    if(value.getClass().isArray()){
+                        jsonGenerator.writeArrayFieldStart(key);
+                        for(String property : (String[]) value){
+                            jsonGenerator.writeString(property);
+                        }
+                        jsonGenerator.writeEndArray();
+                    } else {
+                        jsonGenerator.writeStringField(key, value.toString());
```

Comment:
I don't think this is correct as JSON natively supports: String, Boolean, Numbers
Only datetime and spatial has to be formatted accordingly
for spatial we could use geojson (afaik there is an impl in neo4j-server Tx endpoint for serializing spatial types)
for time values

please try out to return differnt types via the tx-http-endpoint

```
curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":{"statement":"return datetime()"}}'
curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":[{"statement":"return datetime()"}]}'
curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":[{"statement":"return date()"}]}'
curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":[{"statement":"return time()"}]}'

curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":[{"statement":"return point({latitude:53.2,longitude:18})"}]}'

curl -i http://localhost:7474/db/data/transaction/commit/ -H accept:application/json -H content-type:application/json -u neo4j:test -d'{"statements":[{"statement":"return localdatetime()"}]}' 
```

enabled with an _optional_ config parameter we could allow to store the type information for each field
ala

"data":...
"data_types": {"name":"string"} ... etc.

Reply:
Hi @jexp , for this one I create a separate issue ok?

Reply:
yes, thanks.